### ISSUE TITLE
Speed-E'z Exchange & Courier Service - Issue with E-Check

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#574]
In GitLab by @kdjstudios on Nov 22, 2017, 12:44

**Submitted by:** "Joann Browne" <joann@speedez.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/4275842

**Server:** Hosted

**Client/Site:** Speed Ez

**Account:** ALL

**Issue:** 

Several months ago support was working on why my E-Check would not work.  I have never heard anything back so today I was trying to get a customer to try it out and she kept getting a failure.  So I ran it on Authorize.net Terminal Server and it went through, but, fails when trying to run it through SABilling.  I really am red-faced that after telling many customers to use their portal all they get is failures.  Please help with this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 22, 2017, 12:48

@smasih @tim.traylor @kace.coady 

The only ticket I could find about echecks with Speed-Ez; is this one #196. Which in summary we added the functionality for Speed-Ez; however it appears they keep updating their Token and Key; but do not update it in SAB. Here is the communciation via helpdesk; in which Jo-Ann does confirm everything is working: http://www.servicedesk.answernet.com/profiles/ticket/2017-03-09-51184/conversation

Please look into this right away and see why they are not able to process E-Checks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 22, 2017, 15:43

My response to Jo Ann:

Thank you for bringing this to our attention and we have informed the support team to look further into this.

I do recall back in April is when we implemented the E-Check feature for you and that the main issue we had there was that there was a authentication error due to the token and key being changed in Authorize.net, but not being updated in SA Billing. The last email thread response (on 4/28/17) that I was able to locate was you confirming a successful payment. Just a reminder, incase this may be the issue. If you request a token and key in Authorize.net; they generate a new one and you will need to have that updated in SA Billing on the site edit page. This is just what I was able to retrieve and recall from the last interaction about the E-Check functionality. Please let us know if there was another issue reported that I am unable to locate. IN summary we wanted to inform you, until the support team is able to research the issue at hand.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 23, 2017, 24:53

@kdjstudios 

We need some valid bank account to test the e-check payments. Because it's working with sandbox.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 24, 2017, 07:59

@pchaudhary 

There is a valid bank account posted in #196 I believe; however it is Jo-Ann's. But do we not log any of the errors associated with payments in SAB; is that not what we should be checking first?

@tim.traylor @kace.coady 

Do we have a bank account for the Aavaz team to use?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 24, 2017, 09:42

@smasih @tim.traylor 

I reached out to Jo Ann for an account that they had failed echecks process to hopefully have us be able to determine further what is going on. Account 5006 is their test account. Could we please check the failed e-check payments on their account and see what is going on. 

Tim and I tried to view the receipts on those failed e-checks and they are redirecting to the clients portal"https://scc.sabilling.com/clients/sign_in". We were able to find a work around to download the receipt for the last failed echeck, however it seems to be missing a lot of information: [receipt__10_.txt](/uploads/81b47f30ff7a9583633eab6df7598ed9/receipt__10_.txt) 

In summary we need to know why these are failing in more information then what is included on the receipt.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 27, 2017, 01:17

> Tim and I tried to view the receipts on those failed e-checks and they are redirecting to the clients

Fixed this issue.

We made a test payment of $1 using SpeedEz site api keys.

![Screenshot-4](/uploads/5c8e9955eedab4c6d4c478767e8a005b/Screenshot-4.png)

We received an authorize error code **E00007**

![Screenshot-5](/uploads/42a821d1f4ec2207f8165702f4908c83/Screenshot-5.png)


/spend 1h

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Nov 27, 2017, 06:31

* @tim.traylor or @kdjstudios may help in arranging the new API key, since Jo Ann might have changed the API key.

* In the mean while, @smasih will investigate into on how can we provide more information about failed echecks, as we do have the more information available on the backend.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Nov 27, 2017, 06:31

changed time estimate to 1h 30m

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 08:36

@abakshi @smasih 

Thank you for confirming what the issue was. Below is my response to Jo Ann.

The support team was able to look further into the failed transactions and also test a new transaction. They all failed due to "invalid authentication values". This means that the API Key and Token do not match what is in Authorize.net. We have a link below describing how Authorize.net token and keys expire once you request a new key and token. Also to note, if you request a new Key and Token in Authorize.net; you need to update the Key and Token in SA Billing to the latest one in Authorize.net. We have provided detailed instructions below to inform you where you need to update the Key and Token in SA Billing. 

Link to Authorize.net's request and expiration of Keys and Tokens: https://support.authorize.net/authkb/index?page=content&id=A405

Where to update the Key and Token in SA Billing:

1. Login to SAB
2. Click "Display Site Dashboard"
3. Select Speed-Ez's
4. Click "Edit Site"
5. In the section labeled "E-Check Gateway": Update the Key and Token to the latest requested ones in Authorize.net
6. Scroll to the bottom of the page and select "Save"

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 09:25

@smasih @abakshi @tim.traylor 

Response from Jo Ann:

I did change it appropriately like I was supposed to and everything still failed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 09:30

My response to Jo Ann: (As I did not see an updated Key and Token under the site, and I do not see any new failed transactions under the 5006 account she specified)

Hello Jo Ann,

I would just like to confirm that today you requested an updated Key and Token in Authorize.net and then you updated the Key and Token under the site edit page in SA Billing. After saving the edits; you then performed a test charge under which account today? We would like to check the accounts transaction to see if the error has changed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 09:32

Her response:

It was not today it was Friday when I get to my office I will change the key again and then try it

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 14:26

Update from Jo Ann:

Just now tried again and it failed. I am going to use authorize.net terminal server. This is crazy.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 14:26

![IMG_5783](/uploads/438d41758692a34aec16aa02e8f38c82/IMG_5783.jpg)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 14:28

[receipt__11_.txt](/uploads/c9c62f6d5f5e50dab5d69eab7543c83c/receipt__11_.txt)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2017, 14:32

@smasih @tim.traylor 

I can confirm that there is a new Token/Key in SAB now. Please research why Jo Ann is still unable to process any E-Checks. This is extremely important to resolve.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 28, 2017, 24:47

updating priority

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 28, 2017, 05:08

@kdjstudios @tim.traylor 

First I made a payment through sandbox, and I was able to process the payment successfully.

I made two transactions from Joe Ann's Account which she provided in closed [ticket - 196](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/196#note_3618). We needed a valid bank account.

1. 1^st Transaction of a penny from backend which went through.
1. 2^nd Transaction of a penny from UI which also went through.

![Screenshot-3](/uploads/4ed48f466d7a78eb5e1bc4d9d64e5783/Screenshot-3.png)


**As per the code failed echeck transaction is getting all the details, but for some transactions it's not. I just made a test transaction with invalid acc # and routing # and all the details are present in the recipt**

![Screenshot-4](/uploads/2de6a7d9b28eeab6ebda089df7416f0b/Screenshot-4.png)

#### If still, Joe is not able to process the payment, please do let us know we can try it on our end. 

We will stay available on WhatsApp.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 28, 2017, 08:22

@smasih 

Thank you for the update, I have just informed Jo Ann and will keep any immediate updates in WhatsApp if she is still unable to run a successful transaction.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 29, 2017, 04:13

any update fro Joe ..?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 29, 2017, 08:20

@smasih 

Jo Ann, was able to confirm making a successful payment. She found it suspicious that we made no changes. I informed her we would be looking into the failed transaction details amongst other areas to see why it did not work after she immediately changed the Token and Key. 

Per @tim.traylor request too, were you able to find out if there is a time frame that needs to be waited before processing a transaction after updating the Token and Key?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 30, 2017, 04:15

This is what I got from Authorize.net. Means if we are not choosing the option to disable the old key on creating the new one the olde key auto-expires after 24 hr.

**I was wrong on the part that it takes 24hr for a new key to get activated in order to make transactions. New key is immediately activated**

![Screenshot](/uploads/1c5a16f12c445d2e55cc516bd684483d/Screenshot.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 30, 2017, 08:36

@smasih 

So if this was activated immediately, then the transaction that Jo Ann processed right after she changed the Key should have went through successfully, right? Or is there possibly a delay in the SA Billing System?

Or was there any changes the Aavaz team made after she changed her key?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 30, 2017, 23:45

@kdjstudios 

> Or was there any changes the Aavaz team made after she changed her key?


We didn't make any changes to the code file except the debugging print lines to check if everything was working.



> So if this was activated immediately, then the transaction that Jo Ann processed right after she changed the Key should have went through successfully, right? Or is there possibly a delay in the SA Billing System?

There is no delay in SAB and neither in Authorize(*as per documentation there are people who dynamically create new set of keys on every transaction for security purpose.*). There is a possibility that details Joe typed in were incorrect.

Long story short, there is no delay on both sides on changing keys.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 1, 2017, 08:12

@tim.traylor 

With your approval we can close this ticket.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 20, 2018, 04:04

There is nothing left on this, it was just a user error. We should close this one

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 22, 2018, 09:43

marking for close

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 11, 2018, 13:11

closed
